crates/ty_python_semantic/resources/mdtest/type_qualifiers/final.md

```diff
@@ -231,7 +227,86 @@ FINAL_LIST: Final[list[int]] = [1, 2, 3]
 FINAL_LIST[0] = 4
 ```
 
-## Too many arguments
+## Overriding in subclasses
+
+When a symbol is qualified with `Final` in a class, it cannot be overridden in subclasses.
+
+```py
+from typing import Final
+
+class Base:
+    FINAL_A: Final[int] = 1
+    FINAL_B: Final[int] = 1
+    FINAL_C: Final = 1
+
+class Derived(Base):
+    # TODO: This should be an error
+    FINAL_A = 2
+    # TODO: This should be an error
+    FINAL_B: Final[int] = 2
+    # TODO: This should be an error
+    FINAL_C = 2
+```
+
+## Syntax and usage
+
+### Legal syntactical positions
+
+Final may only be used in assignments or variable annotations. Using it in any other position is an
+error.
+
+```py
+from typing import Final, ClassVar, Annotated
+
+LEGAL_A: Final[int] = 1
+LEGAL_B: Final = 1
+LEGAL_C: Final[int]
+LEGAL_C = 1
+LEGAL_D: Final
+LEGAL_D = 1
+
+class C:
+    LEGAL_E: ClassVar[Final[int]] = 1
+    LEGAL_F: Final[ClassVar[int]] = 1
+    LEGAL_G: Annotated[Final[ClassVar[int]], "metadata"] = 1
+
+    def __init__(self):
+        self.LEGAL_H: Final[int] = 1
+        self.LEGAL_I: Final[int]
+        self.LEGAL_I = 1
+
+# TODO: This should be an error
+def f(ILLEGAL: Final[int]) -> None:
+    pass
+```
+
+### Attribute assignment outside `__init__`
+
+Qualifying an instance attribute with `Final` outside of `__init__` is not allowed. The instance
+attribute must be assigned only once, when the instance is created.
+
+```py
+from typing import Final
+
+class C:
+    def some_method(self):
+        # TODO: This should be an error
+        self.x: Final[int] = 1
+```
+
+### `Final` in loops
+
+Using `Final` in a loop is not allowed.
+
+```py
+from typing import Final
+
+for i in range(10):
+    # TODO: This should be an error
+    i: Final[int] = 1
```

Comment:
I think it would be clearer here to use a different name for the loop variable -- the declaration in the loop body is illegal because the variable would be repeatedly assigned on each iteration of the loop, not because it shadows the loop variable:

```suggestion
for _ in range(10):
    # TODO: This should be an error
    i: Final[int] = 1
```

Reply:
Oops, yes!